### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by eliottparis**
This PR adds necessary files for the initial release of the *flip-lang-vscode* extension.

### CODE REVIEW

File: syntaxes/flip.tmLanguage.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            {
+               "name": "keyword.other.flip",
+               "match": "\\b(signal|typealias)\\b"
+            }
+         ]
+      },
+      "storages": {
+         "patterns": [
+            {
+               "name": "storage.type.flip",
```
Comment (ohmtech-rdi) [inline syntaxes/flip.tmLanguage.json]: I would say those are keywords as well, not types

File: syntaxes/flip.tmLanguage.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+               "match": "\\b(model|class|enum|invariant)\\b"
+            },
+            {
+               "name": "storage.type.flip",
+               "match": "\\b(Array|Bool|Blob|Collection|Float|Int|Message|Map|ObjectRef|Optional|String|Variant|Vector)\\b"
+            },
+            {
+               "name": "storage.modifier.flip",
```
Comment (ohmtech-rdi) [inline syntaxes/flip.tmLanguage.json]: Ok with that if VSCode users would expect that (or the same can be find in Java/Javascript, etc)

**Review by ohmtech-rdi — CHANGES_REQUESTED** [review]
Apart from the 2 points mentioned, the rest looks good to me at first sight (I didn't get into the details of VSCode, so I'll trust you on that)

File: syntaxes/flip.tmLanguage.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+   "repository": {
+      "keywords": {
+         "patterns": [
+            {
+               "name": "keyword.control.flip",
+               "match": "\\b(return)\\b"
+            },
+            {
+               "name": "keyword.other.flip",
+               "match": "\\b(signal|typealias)\\b"
```
Comment (ohmtech-rdi) [inline syntaxes/flip.tmLanguage.json]: Please move `signal` with `invariant`

**Review by ohmtech-rdi — COMMENTED** [review]
Apart from that, I would say that it might be nice to specify the `storage.type`name '(like `storage.type.function` for `signal` and `invariant`), but if this doesn't play well with VSCode, then it's good for me

File: syntaxes/flip.tmLanguage.json (hunk 3 above)
Comment (eliottparis) [reply syntaxes/flip.tmLanguage.json]: Ok, I will do that

**Comment by eliottparis** [comment]
@ohmtech-rdi, I updated the syntax grammar file following your comments to be more specific on the names we give to each token. 

Next step would be to introduce `meta`s to be more specific on the context in which each token can be used. But let's do that in another PR/version of the extension.

**Review by ohmtech-rdi — APPROVED** [review]

**Pull request merged**